From 43ec50c9cf7393cbb7d969aef8c9d4afd36ca779 Mon Sep 17 00:00:00 2001
From: Aleksandr Semenov <smallcsf@gmail.com>
Date: Mon, 17 Apr 2023 20:41:01 +0300
Subject: [PATCH] Mark include as system
---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9f583af..7ba850a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -81,7 +81,7 @@ else()
     APPEND_STRING PROPERTY COMPILE_FLAGS " -O3")
 endif()
 
-target_include_directories(rapidcheck PUBLIC
+target_include_directories(rapidcheck SYSTEM PRIVATE
     $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include/>
     $<INSTALL_INTERFACE:include>  # <prefix>/include
 )
-- 
2.25.1

