From 6167e7a37556a84033becc5b4fb5ccbf7858630d Mon Sep 17 00:00:00 2001
From: vagin ivan <vagin.ivan1@huawei-partners.com>
Date: Wed, 13 Jul 2022 12:36:24 +0300
Subject: [PATCH] Migrate from clang-9 to clang-14

Signed-off-by: vagin ivan <vagin.ivan1@huawei-partners.com>
---
 CMakeLists.txt | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 37a1593..7d352b4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -166,6 +166,13 @@ if (NOT ASMJIT_NO_CUSTOM_FLAGS)
     #list(APPEND ASMJIT_PRIVATE_CFLAGS -Wall -Wextra -Wconversion)
     list(APPEND ASMJIT_PRIVATE_CFLAGS -fno-math-errno)
     list(APPEND ASMJIT_PRIVATE_CFLAGS -Wno-shadow)
+
+    if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 14.0.0)
+      list(APPEND ASMJIT_PRIVATE_CFLAGS -Wno-bitwise-instead-of-logical)
+      list(APPEND ASMJIT_PRIVATE_CFLAGS -Wno-unused-but-set-variable)
+      list(APPEND ASMJIT_PRIVATE_CFLAGS -Wno-deprecated-copy)
+    endif()
+
     list(APPEND ASMJIT_PRIVATE_CFLAGS_REL -O2)
 
     asmjit_detect_cflags(ASMJIT_PRIVATE_CFLAGS
-- 
2.25.1

